### ISSUE TITLE
react-native-maps and beta Google map service shutdown

### ISSUE THREAD

**Issue by soullivaneuh** [expo/expo#16298]
### Summary

I recently received an email from Google Maps telling the current beta service will be closed at November 2022.

I use `react-native-maps`, that is managed by `expo install`, and it has an opened issue about that: https://github.com/react-native-maps/react-native-maps/issues/4087

For now, we don't have any answer of the maintainers.

This is not directly managed by you, but it may/will have a big impact. At least, you are informed if not. :+1: 

Do we have any way to workaround this issue on the Expo side? :thinking: 

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### Environment

```
  Expo CLI 4.12.0 environment info:
    System:
      OS: Linux 5.10 Manjaro Linux
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 17.3.0 - /usr/bin/node
      Yarn: 1.22.17 - /usr/bin/yarn
      npm: 8.4.0 - /usr/bin/npm
    npmPackages:
      expo: ^42.0.0 => 42.0.1 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: https://github.com/expo/react-native/archive/sdk-42.0.0.tar.gz => 0.63.2 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.12.0
    Expo Workflow: managed
```

### Reproducible demo

N/A

**Comment by brentvatne** [comment]
thanks for the heads up! we'll monitor this situation

**Comment by jsjexpert** [comment]
There's a new branch available for react-native-maps which allows for using the new renderer in SDK 18:
[https://github.com/react-native-maps/react-native-maps/tree/beta](https://github.com/react-native-maps/react-native-maps/tree/beta)
[https://github.com/react-native-maps/react-native-maps/commit/bb0855714895e5264f4f6b3ebc9f3cdc6350b39a](https://github.com/react-native-maps/react-native-maps/commit/bb0855714895e5264f4f6b3ebc9f3cdc6350b39a)

Available on npm as react-native-maps@0.31.0-beta.3

I haven't had success using it in Expo as i get a gradle error when trying to build with EAS:

`[stderr] /home/expo/workingdir/build/node_modules/react-native-maps/lib/android/src/main/java/com/airbnb/android/react/maps/AirMapModule.java:368: error: lambda expressions are not supported in -source 1.7
[stderr]         MapsInitializer.initialize(context, MapsInitializer.Renderer.LATEST, (MapsInitializer.Renderer renderer) -> {
[stderr]                                                                                                                  ^
[stderr]   (use -source 8 or higher to enable lambda expressions)
[stderr] 1 error`

**Comment by soullivaneuh** [comment]
New rendered seems releases since [v0.31.0-beta.2](https://github.com/react-native-maps/react-native-maps/releases/tag/v0.31.0-beta.2) according to https://github.com/react-native-maps/react-native-maps/issues/3991#issuecomment-1079436285.

**Comment by soullivaneuh** [comment]
@brentvatne I tried `0.31.0-beta.2` with `expo start`, looks to work correctly. However, it look the build is not working according to @jsjexpert.

May plan a dependency compatibility update to resolve this issue?

Thanks!

**Comment by jsjexpert** [comment]
@soullivaneuh The build issue was resolved on the Expo side with react-native-maps 0.31.1. Make sure you are using expo to install instead of yarn or npm and it will now build correctly

**Comment by soullivaneuh** [comment]
@jsjexpert I did run the following commands:

```
npm remove -f react-native-maps && expo install react-native-maps -- -f
```

And it ends with version `0.30.2` of the package.

If I manually set it to `0.31.1` on `package.json`, it works, but the expo cli trigger a warning about the version compatibility:

```
Some dependencies are incompatible with the installed expo version:
  react-native-maps@0.31.1 - expected version: 0.30.2
Your project may not work correctly until you install the correct versions of the packages.
Install individual packages by running npx expo install react-native-maps@0.30.2
```

My `expo-cli` version is `5.4.12`.

Did I miss something from the process? :thinking:

**Comment by jsjexpert** [comment]
It's been almost two months since I ran this build, but I've got expo-cli 5.3.0

From my package.json
`{
  "scripts": {
    "start": "expo start --dev-client",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "web": "expo start --web",
    "eject": "expo eject"
  },
  "dependencies": {
    "@expo-google-fonts/inter": "^0.1.0",
    "@expo-google-fonts/roboto": "^0.1.0",
    "@react-native-async-storage/async-storage": "~1.15.0",
    "@react-native-community/masked-view": "0.1.10",
    "@react-navigation/native": "^5.8.10",
    "@react-navigation/stack": "^5.12.8",
    "axios": "^0.21.1",
    "expo": "^44.0.0",
    "expo-app-loading": "~1.3.0",
    "expo-application": "~4.0.1",
    "expo-constants": "~13.0.0",
    "expo-dev-client": "~0.8.4",
    "expo-font": "~10.0.4",
    "expo-linear-gradient": "~11.0.0",
    "expo-linking": "~3.0.0",
    "expo-localization": "~12.0.0",
    "expo-location": "~14.0.1",
    "expo-navigation-bar": "~1.1.1",
    "expo-splash-screen": "~0.14.1",
    "expo-status-bar": "~1.2.0",
    "expo-system-ui": "~1.1.0",
    "i18next": "^19.8.4",
    "react": "17.0.1",
    "react-dom": "17.0.1",
    "react-i18next": "^11.8.5",
    "react-native": "0.64.3",
    "react-native-gesture-handler": "~2.1.0",
    "react-native-locale-detector": "^1.0.1",
    "react-native-maps": "^0.31.1",
    "react-native-reanimated": "~2.3.1",
    "react-native-safe-area-context": "3.3.2",
    "react-native-screens": "~3.10.1",
    "react-native-snap-carousel": "^3.9.1",
    "react-native-svg": "12.1.1",
    "react-native-web": "0.17.1",
    "react-native-web-hooks": "^3.0.1",
    "striptags": "^3.2.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@types/googlemaps": "^3.43.2",
    "@types/react": "~17.0.21",
    "@types/react-dom": "~17.0.9",
    "@types/react-native": "~0.64.12",
    "@types/react-native-snap-carousel": "^3.8.5",
    "typescript": "~4.3.5"
  },
  "private": true,
  "name": "zzz",
  "version": "0.0.0"
}`

The fixes were released here: https://github.com/react-native-maps/react-native-maps/releases/tag/v0.31.0
You're right though, I thought I had used expo install but it's still set to version 0.30.2.

**Comment by tanifort** [comment]
I ran into a problem trying to use react-native-maps with expo; The map won't show up, it displays a blank screen. The same blank screen shows when I try to run the _react-native-maps_ example provided on the expo website. I'm stuck since **MapView** is the most important component of my app. 
Is there any workaround??

**Comment by byCedric** [comment]
@tanifort [You are describing issue #18554](https://github.com/expo/expo/issues/18554#issuecomment-1216561781), which is unrelated to this issue. This one is to keep an eye on Google's deprecation notice of older Google Maps SDKs. Fix for that should be out already.

**Comment by tanifort** [comment]
You are right. I have switched back to react native cli and everything works as expected. I had been away from expo because of this kind of support issues and I think that i will just stay away from it.

**Comment by brentvatne** [comment]
sdk 47 includes react-native-maps 1.3.2

**Comment by soullivaneuh** [comment]
@brentvatne Not sure you understand my feedback: The new Google service usage is available ficne v0.31.0-betra.2 of react-native-maps, and an higher version **looks already integrated starting SDK v46**.

So I have two questions now ha ha :

1. Is the issue problem considered as resolved and be safely closed?
1. You are talking about SDK 47, but do we agree having v46 is already enough to fix this issue?

Thanks!

**Comment by vinaciotm** [comment]
🎉 SOLUTION: (USING EXPO SDK 47.0.8) 🎉

UNDERSTADING THE PROBLEM:
- Development environment use debug SHA1
- Production environment use appSigning SHA1
- Your API KEY in CLOUD have permission for Production, but not for Debug.

STEP 1: in your expo project, run `npx expo run:android`
STEP 2: run `cd android`
STEP 3: run `gradle signingReport`
STEP 4: Look for `> Task :app:signingReport` with text `/android/app/debug.keystore` you will see your debug keystore, look for `SHA1` (this is the debug SHA1 that you need to insert on cloud)

-----------------------------------------

* POSSIBLE PROBLEM AND HOW TO FIX:
** Look for `> Task :expo:signingReport` with text `Error: Missing keystore` if is missing, do this:
--STEP A: COPY the `debug.keystore` file in `YOUR_ROOT_PROJECT/android/app/debug.keystore`
--STEP B: run `open /Users/inaciopires/.android/` and PASTE the `debug.keystore` file
--STEP C: lets, check, run again `gradle signingReport` and check if your keystore is not missing anymore.
-----------------------------------------

STEP 5: create a new API KEY for dev debug in https://console.cloud.google.com/google/maps-apis/credentials
and add your `SHA1` that you find in `STEP 4`.

🎉 CONGRATULATIONS 🎉 
Now your development debug environment will works again

IF YOUR PRODUCTION OR INTERNAL TEST NOT WORKING
- STEP A: open the console https://play.google.com/console/u/1/developers/ and select your App
- STEP B: Search for `App Integrity` and select the tab `App sign`
- STEP C:  copy the `SHA-1 certification`
- STEP D: follow the `STEP 5` and create your API KEY for this SHA1

🎉  CONGRATULATIONS 🎉 
Now your production environment will works

🇧🇷 Thanks, I hope I helped 😉
